clients/include/testing_rotm_batched.hpp

```diff
@@ -0,0 +1,274 @@
+    if(!dx || !dy || !dparam)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_rotm_batched<T>(nullptr, N, dx, incx, dy, incy, dparam, batch_count)),
+        rocblas_status_invalid_handle);
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_rotm_batched<T>(handle, N, nullptr, incx, dy, incy, dparam, batch_count)),
+        rocblas_status_invalid_pointer);
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_rotm_batched<T>(handle, N, dx, incx, nullptr, incy, dparam, batch_count)),
+        rocblas_status_invalid_pointer);
+    EXPECT_ROCBLAS_STATUS(
+        (rocblas_rotm_batched<T>(handle, N, dx, incx, dy, incy, nullptr, batch_count)),
+        rocblas_status_invalid_pointer);
+}
+
+template <typename T>
+void testing_rotm_batched(const Arguments& arg)
+{
+    rocblas_int N           = arg.N;
+    rocblas_int incx        = arg.incx;
+    rocblas_int incy        = arg.incy;
+    rocblas_int batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+    double               gpu_time_used, cpu_time_used;
+    double norm_error_host_x = 0.0, norm_error_host_y = 0.0, norm_error_device_x = 0.0,
+           norm_error_device_y = 0.0;
+
+    // check to prevent undefined memory allocation error
+    if(N <= 0 || incx <= 0 || incy <= 0 || batch_count <= 0)
+    {
+        static const size_t     safe_size = 100; // arbitrarily set to 100
+        device_vector<T*, 0, T> dx(safe_size);
+        device_vector<T*, 0, T> dy(safe_size);
+        device_vector<T>        dparam(5);
+        if(!dx || !dy || !dparam)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_device));
+        if(batch_count < 0)
+            EXPECT_ROCBLAS_STATUS(
+                (rocblas_rotm_batched<T>(handle, N, dx, incx, dy, incy, dparam, batch_count)),
+                rocblas_status_invalid_size);
+        else
+            CHECK_ROCBLAS_ERROR(
+                (rocblas_rotm_batched<T>(handle, N, dx, incx, dy, incy, dparam, batch_count)));
+        return;
+    }
+
+    size_t size_x = N * size_t(incx);
+    size_t size_y = N * size_t(incy);
+
+    device_vector<T*, 0, T> dx(batch_count);
+    device_vector<T*, 0, T> dy(batch_count);
+    device_vector<T>        dparam(5);
+
+    if(!dx || !dy || !dparam)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    host_vector<T> hx[batch_count];
+    host_vector<T> hy[batch_count];
+    host_vector<T> hdata(4);
+    host_vector<T> hparam(5);
+
+    device_batch_vector<T> bx(batch_count, size_x);
+    device_batch_vector<T> by(batch_count, size_y);
+
+    for(int b = 0; b < batch_count; b++)
+    {
+        hx[b] = host_vector<T>(size_x);
+        hy[b] = host_vector<T>(size_y);
+    }
+
+    int last = batch_count - 1;
+    if((!bx[last] && size_x) || (!by[last] && size_y))
```

Comment:
Probably need to make batch vector fail if any allocation fail to avoid such things as this last test.

Reply:
Hm, I'm not sure I follow what you mean. This is the check that was done for device_batch_vectors in the original gemv_batched so I've been following that.

Reply:
One shouldn't have to check one of the list for failure, the allocator should return failure if any element of a vector fails to allocate.  This would be addressed in a new jira issue and PR.   Not sure if you are keen to create the issue.   It could be considered technical debt.  Can resolve this conversation.

Reply:
Ah, I see. Iirc Yvan is looking at improving device_vector and device_batch_vector? Could be brought to his attention.

Reply:
Sure I will fwd this to him to add to the issue he created.